src/pip/_internal/models/installation_report.py

```diff
@@ -23,6 +23,9 @@ def _install_req_to_dict(cls, ireq: InstallRequirement) -> Dict[str, Any]:
             # includes editable requirements), and false if the requirement was
             # downloaded from a PEP 503 index or --find-links.
             "is_direct": ireq.is_direct,
+            # is_yanked is true if the requirement was yanked from the index, but
+            # was still selected by pip to conform to PEP 592.
+            "is_yanked": ireq.link.is_yanked if ireq.link else False,
```

Comment:
Do we know in which circumstances `ireq.link` can be None ?

Perhaps we should output `"is_yanked": None` in that case, to express that we don't know if it's yanked or not, rather the False which asserts it is not yanked.

Reply:
when `ireq.link` is None, it's not coming from a pypi index. since the notion of yanking only applies to pypi indices, I think returning False here is safe. but please correct me if I'm wrong @pfmoore

Reply:
> when ireq.link is None, it's not coming from a pypi index

If that's the reason of ireq.link is None, then it's fine to return is_yanked=False in such case, indeed.